File: codeforces/practice/800/263A-beautiful-matrix.cpp
#include <bits/stdc++.h>
using namespace std;
int main() {
    for (int i = 0; i < 5; i++)        // loop through each row
        for (int j = 0; j < 5; j++) {  // loop through each column
            int x; cin >> x;            // read current cell value
            if (x == 1)                 // found the 1
                // minimum moves = manhattan distance to centre (index 2,2)
                cout << abs(i - 2) + abs(j - 2);
        }
}

File: codeforces/practice/800/263A-beautiful-matrix.py
# Read the 5x5 matrix and find the position of the single 1
for i in range(5):
    row = input().split()  # read each row as a list of strings
    for j in range(5):
        if row[j] == '1':  # found the 1
            # minimum moves = manhattan distance to centre (row 3, col 3 = index 2,2)
            print(abs(i - 2) + abs(j - 2))